### PR TITLE
Final or polymorphic

### PR DESCRIPTION
**Pull request by eggrobin**
Address #596.
Some virtual destructors were missing!
Removed useless non-virtual default destructors.

### CODE REVIEW

File: quantities/quantities_body.hpp [hunk 1]
```diff
@@ -117,13 +115,11 @@ struct QuotientGenerator {
                           LuminousIntensity, Angle>>>::Type;
 };
 template<typename Left>
-struct QuotientGenerator<Left, double> { using Type = Left; };
+struct QuotientGenerator<Left, double> : type_trait { using Type = Left; };
```
Comment (pleroy) [inline quantities/quantities_body.hpp]: Line break after `{`.

File: quantities/quantities_body.hpp [hunk 2]
```diff
@@ -117,13 +115,11 @@ struct QuotientGenerator {
...
 template<>
-struct QuotientGenerator<double, double> {
-  using Type = double;
-};
+struct QuotientGenerator<double, double> : type_trait { using Type = double; };
```
Comment (pleroy) [inline quantities/quantities_body.hpp]: Same.

File: suppress_useless_warnings.props [hunk 3]
```diff
@@ -5,7 +5,7 @@
   <PropertyGroup />
   <ItemDefinitionGroup>
     <ClCompile>
-      <DisableSpecificWarnings>4018;4373;4503;4554;4566;4996;4244;4267;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <DisableSpecificWarnings>4018;4373;4503;4554;4566;4996;4244;4267;4624%(DisableSpecificWarnings)</DisableSpecificWarnings>
```
Comment (pleroy) [inline suppress_useless_warnings.props]: Can we sort these numbers?

File: tools/journal_proto_processor.cpp [hunk 4]
```diff
@@ -904,7 +904,7 @@ void JournalProtoProcessor::ProcessMethodExtension(
   std::string cxx_interface_return_type = "void";
   std::string cxx_run_prolog;
   std::string cxx_run_epilog;
-  cxx_toplevel_type_declaration_[descriptor] = "struct " + name + " {\n";
+  cxx_toplevel_type_declaration_[descriptor] = "struct " + name + " : type_trait {\n";
```
Comment (pleroy) [inline tools/journal_proto_processor.cpp]: Let's explicitly delete the destructor.

File: quantities/quantities_body.hpp (hunk 1 above)
Comment (eggrobin) [inline quantities/quantities_body.hpp]: Done.

File: quantities/quantities_body.hpp (hunk 2 above)
Comment (eggrobin) [inline quantities/quantities_body.hpp]: Done.

File: suppress_useless_warnings.props (hunk 3 above)
Comment (eggrobin) [inline suppress_useless_warnings.props]: Done.

File: tools/journal_proto_processor.cpp (hunk 4 above)
Comment (eggrobin) [inline tools/journal_proto_processor.cpp]: As discussed, it's better this way.

File: journal/method.hpp [hunk 5]
```diff
@@ -5,18 +5,18 @@
 #include <memory>
 
 #include "base/not_null.hpp"
-#include "base/type_traits.hpp"
+#include "base/not_constructible.hpp"
```
Comment (pleroy) [inline journal/method.hpp]: Alphabetize.

File: journal/profiles.hpp [hunk 6]
```diff
@@ -2,15 +2,15 @@
 #pragma once
 
 #include "base/not_null.hpp"
-#include "base/type_traits.hpp"
+#include "base/not_constructible.hpp"
```
Comment (pleroy) [inline journal/profiles.hpp]: Same.  Please check them all.

**Pull request merged**